core/src/main/java/de/danoeh/antennapod/core/service/download/NewEpisodesNotification.java

```diff
@@ -0,0 +1,94 @@
+package de.danoeh.antennapod.core.service.download;
+
+import android.app.Notification;
+import android.app.PendingIntent;
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.Intent;
+import android.content.res.Resources;
+
+import androidx.core.app.NotificationCompat;
+import androidx.core.app.NotificationManagerCompat;
+
+import de.danoeh.antennapod.core.R;
+import de.danoeh.antennapod.core.feed.Feed;
+import de.danoeh.antennapod.core.feed.FeedPreferences;
+import de.danoeh.antennapod.core.preferences.UserPreferences;
+import de.danoeh.antennapod.core.storage.DBReader;
+import de.danoeh.antennapod.core.storage.PodDBAdapter;
+import de.danoeh.antennapod.core.util.gui.NotificationUtils;
+
+public class NewEpisodesNotification {
+    static final String GROUP_KEY = "de.danoeh.antennapod.EPISODES";
+
+    private final int lastEpisodeCount;
+    private final boolean shouldShowNotification;
+
+    public NewEpisodesNotification(Long feedId) {
+        Feed feed = DBReader.getFeed(feedId);
```

Comment:
> Maybe I would still use the commit from notify-change-pref-cursor because at the moment we unesarally fetch feed info like LastModifiedDate, author etc.

Oh, I was not aware that `DBReader.getFeed` also loads all items of the feed. I don't think author etc are a problem - they don't need much space. The whole list of feed items definitely does, though. I think it would be good to add a method `DBReader.getFeedWithoutItems` that can then also be used by other classes. The method with a `Cursor` does not decouple the classes enough, in my opinion. I think notification code should not know anything about database internals.